app/src/main/java/com/github/lookupgroup27/lookup/model/map/renderables/utils/GeometryUtils.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.lookupgroup27.lookup.model.map.renderables.utils
+
+/**
+ * Utility object for generating spherical geometry. This class provides methods for creating vertex
+ * data, color data, texture coordinates, and indices for rendering a 3D sphere using OpenGL.
+ *
+ * The sphere is tessellated into latitude bands and longitude steps, which control the level of
+ * detail and smoothness of the sphere.
+ */
+object Sphere {
+  fun generateSphericalGeometry(numBands: Int = 20, stepsPerBand: Int = 28): GeometryData {
+    val vertices = mutableListOf<Float>()
+    val colors = mutableListOf<Int>()
```

Comment:
I’m not entirely sure, but I think we might not need `color` anymore since we’re now consistently using a texture. It might be worth considering whether we can remove this to simplify the code. What do you think?

Reply:
You're right, thank you for noticing!